bubblewrap.c

```diff
@@ -1327,6 +1333,10 @@ main (int    argc,
   if (!is_privileged)
     opt_unshare_user = TRUE;
 
+  if (opt_unshare_user_try &&
+      stat ("/proc/self/ns/user", &sbuf) == 0)
```

Comment:
I think my vote would be detecting CentOS/RHEL in `configure.ac` and doing `#ifdef RHEL have_userns = FALSE #else have_users = stat(...) == 0 #endif` or so.